src/_pytest/unittest.py

```diff
@@ -396,7 +396,7 @@ def excstore(
     return res
 
 
-def check_testcase_implements_trial_reporter(done: List[int] = []) -> None:
+def check_testcase_implements_trial_reporter(done: List[int] = []) -> None:  # noqa: B006
```

Comment:
Ooh this is pretty terrible, I suggest we refactor the code to avoid this hidden global state.

Reply:
This one I didn't change yet, I suppose the intent was to do this once instead of multiple time:
```
    from twisted.trial.itrial import IReporter
    from zope.interface import classImplements

    classImplements(TestCaseFunction, IReporter)
```
When launching 
```
@hookimpl(wrapper=True)
def pytest_runtest_protocol(item: Item) -> Generator[None, object, object]:
    if isinstance(item, TestCaseFunction) and "twisted.trial.unittest" in sys.modules:
        ut: Any = sys.modules["twisted.python.failure"]
        Failure__init__ = ut.Failure.__init__
        check_testcase_implements_trial_reporter()
```
I suppose it's a costly operation. Any ideas how to fix or should I just use a more "politically correct" global state ?

Reply:
Yes, let's at least not "hide" the global var in the parameter list, I think it's a bad pattern. Perhaps we will deal with the actual global sometime...

Reply:
So should I fix this in a subsequent merge request or in this one ?

Reply:
I prefer not to add noqa for real problems so either fix now (move `done` to a global var rather than a mutable default arg) or ignore B006 for the project and fix later.

Reply:
I used an existing global state from python (8c851441489d855fb7e1a0c34230927b5a1c9a08 / https://docs.python.org/3/library/functions.html#dir) let me know what you think.
```python
>>> import sys
>>> "sys" in sys.modules
True
>>> "version_info" in dir()
False
>>> from sys import version_info
>>> "version_info" in dir()
True
```

Reply:
I might be misunderstand how `dir` works but are sure `"classImplements" not in dir()` is not always false at the point it is called?

Reply:
Right..
```python
import sys
def in_function_scope():
    if not "version_info" in globals():
        from sys import version_info

        print(f"Imported version_info")
    else:
        print("version_info already imported")


in_function_scope()
in_function_scope()
```
=>
```
Imported version_info
Imported version_info
```
I'm going to just use a global

Reply:
```python

import sys

classImplements_has_run = False

def in_function_scope():
    global classImplements_has_run
    if not classImplements_has_run:
        from sys import version_info


        print(f"Imported version_info")
        classImplements_has_run = True
    else:
        print("version_info already imported")





in_function_scope()
in_function_scope()
```


```
Imported version_info
version_info already imported
```